R/utilities-output-selections.R

```diff
@@ -17,24 +20,32 @@
 #' parameter <- getParameter("Organism|Liver|Volume", sim)
 #' addOutputs(parameter, sim)
 #' @export
-addOutputs <- function(quantitiesOrPaths, simulation) {
+addOutputs <- function(quantitiesOrPaths, simulation, returnQuantities = TRUE) {
   quantitiesOrPaths <- c(quantitiesOrPaths)
 
   validateIsOfType(quantitiesOrPaths, c(Quantity, "character"))
   validateIsOfType(simulation, Simulation)
-  quantities <- quantitiesOrPaths
 
-  if (isOfType(quantitiesOrPaths, "character")) {
-    quantities <- getAllQuantitiesMatching(quantitiesOrPaths, simulation)
+  # If quantities are provided, get their paths
+  paths <- vector("character", length(quantitiesOrPaths))
+  if (isOfType(quantitiesOrPaths, Quantity)) {
+    for (idx in seq_along(quantitiesOrPaths)) {
+      paths[[idx]] <- quantitiesOrPaths[[idx]]$path
+    }
+  } else {
+    paths <- quantitiesOrPaths
   }
+  paths <- unique(paths)
 
-  quantities <- uniqueEntities(quantities, compareBy = "path")
-  outputSelections <- simulation$outputSelections
-
-  for (quantity in quantities) {
-    outputSelections$addQuantity(quantity)
+  task <- ospsuite:::getContainerTask()
+  for (path in paths) {
+    rClr::clrCall(task, "AddQuantitiesToSimulationOutputFromPath", simulation$ref, enc2utf8(path))
   }
 
+  quantities <- NULL
+  if (returnQuantities) {

```

Comment:
Same comment as before. Here we are changing return values based on input values (and even more than the other one) which feels very wrong
I do not know if we `really` need to return the quantity from this methods. Maybe it should just have no return values since it has side effects . I 'd rather break this API rather than implementing this kind of logic to be honest

Reply:
Returning the quantities here takes 10 seconds in the steady-state example, which is not acceptable.

So either do not return them at all, or create a separate method (`addOuputsSilent` or something), which I do not like at all..

Reply:
@Yuri05 I would vote for not returning the Outptus. If the user wants them, it's a one liner. I want to avoid complex APIs where not required.